File: Chapter-27/ConvexHull/chull_OL.py
from scipy.spatial import ConvexHull, convex_hull_plot_2d
import matplotlib.pyplot as plt
import numpy as np

locations = np.array([[89.13,31.65],
                        [86.32,26.70],
                        [90.03,28.61],
                        [87.83,31.59],
                        [88.77,28.45],
                        [87.52,27.26]])

# calculate the convex hull
ol_hull = ConvexHull(locations)

print("The area of the OL convex hull is: ", ol_hull.area)


# plot the data

plt.xlim((0,100))
plt.ylim((0,60))
plt.plot(locations[:,0],locations[:,1], 'o')

for simplex in hull.simplices:
        plt.plot(locations[simplex, 0], locations[simplex, 1], 'k-')

plt.plot(locations[hull.vertices,0], locations[hull.vertices,1], 'r--', lw=2)
plt.plot(locations[hull.vertices[0],0], locations[hull.vertices[0],1], 'ro')
plt.show()

File: Chapter-27/ConvexHull/chull_OL.R
library(grDevices)
library(geometry)

locations <- matrix(c(89.13, 31.65,
                      86.32, 26.70,
                      90.03, 28.61,
                      87.83, 31.59,
                      88.77, 28.45,
                      87.52, 27.26), ncol = 2, byrow = TRUE)

# calculate the convex hull
ol_hull <- chull(locations)

ol_area <- polyarea(locations[ol_hull, 1], locations[ol_hull, 2])

print(paste0("The area of the OL convex hull is: ", ol_area))


# plot the data

plot(locations, xlim = c(80, 100), ylim = c(0, 50), xlab = "", ylab = "", asp = 1)
points(locations, pch = 20)

hull_edges <- rbind(locations[ol_hull, ], locations[ol_hull[1], ])
lines(hull_edges, col = "red", lwd = 2)
points(locations[ol_hull[1], ], pch = 19, col = "red")

